Hardcore.lua

```diff
@@ -318,103 +318,12 @@ local function SlashHandler(msg, editbox)
 		for substring in args:gmatch("%S+") do
 			dk_convert_option = substring
 		end
-		--Hardcore:Print(dk_convert_option)
-		local _, _, classID = UnitClass("player")
-		local level = UnitLevel("player")
-		local inCombat = UnitAffectingCombat("player")
-
-		--Hardcore:Print("Class ID: ".. classID .. ", player level: ".. PLAYER_LEVEL)
-
 		if dk_convert_option == "sacrifice" then
-			if inCombat == true then
-				Hardcore:Print("Can't use sacrifice in combat")
-				return
-			end
-			-- check if eligible
-			if classID == 6 then
-				Hardcore:Print("You can't sacrifice Death Knight character")
-				return
-			end
-			if level ~= SACRIFICE_LEVEL then
-				Hardcore:Print(string.format("You must be level %s to sacrifice", SACRIFICE_LEVEL))
-				return
-			end
-			-- need to warn before sacrifice if something is wrong
-			local debug_message = "Playtime gap percentage: " .. Hardcore_Character.tracked_played_percentage .. "%."
-			Hardcore:Debug(debug_message)
-			local percentage = Hardcore_Character.tracked_played_percentage
-
-			if Hardcore:ShouldShowPlaytimeWarning(level, percentage) then
-				Hardcore:DisplayPlaytimeWarning(level)
-				return
-			end
-			-- we deaths, bh, played_time_gaps and trade partners
-			if (#Hardcore_Character.deaths > 0 or #Hardcore_Character.bubble_hearth_incidents > 0 or #Hardcore_Character.played_time_gap_warnings > 0 or #Hardcore_Character.trade_partners > 0) then
-				Hardcore:Print("Before proceeding with sacrifice, please contact admins and verify your character as there are some warnings!")
-				return
-			end
-			local sacrifice = {}
-			sacrifice["guid"] = Hardcore_Character.guid
-			sacrifice["localtime"] = date("%m/%d/%y %H:%M:%S")
-			sacrifice["timestamp"] = time(date("*t"))
-			sacrifice["override"] = false
-			sacrifice["complete"] = false
-			if (Hardcore_Settings.sacrifice == nil) then
-				Hardcore_Settings.sacrifice = {}
-			end
-			for k, v in pairs(Hardcore_Settings.sacrifice) do Hardcore_Settings.sacrifice[k]=nil end
-			table.insert(Hardcore_Settings.sacrifice, sacrifice)
-			--Hardcore_Character.sacrificed_at = sacrifice["localtime"]
-			Hardcore:Print("Character marked for sacrifice. Die within 5 minutes and then activate it on Death Knight.")
+			Hardcore:DKConvert(dk_convert_option) 
```

Comment:
the if, elseif clauses are no longer needed since you are passing the condition as an argument.

Reply:
True. Moved the hint for options into function and those if else is redundant.